tests/modules/nf-core/stitch/main.nf

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { STITCH } from '../../../../modules/nf-core/stitch/main.nf'
+
+// positions and essential parameters
+def posfile         = file(params.test_data['homo_sapiens']['genome']['genome_21_stitch_posfile'], checkIfExists: true)
+def input           = []
+def rdata           = []
+def chromosome_name = "chr21"
+def K               = 2
+def nGen            = 1
+def stitch_input    = [ [ id: "test_positions" ], posfile, input, rdata, chromosome_name, K, nGen ]
+
+// sequencing data
+def crams = [
+    params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram' ],
+    params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_cram'],
+]
+def crais = [
+    params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram_crai' ],
+    params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_cram_crai'],
+]
+def reads = [ [ id:"test_reads" ], crams, crais ]
+
+// reference genome
+def reference = [
+    [ id:"test_reference" ],
+    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta']    , checkIfExists: true),
+    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true),
+]
+
+// for reproducibility
+def seed = 1
+
+
+workflow test_stitch {
+
+    cramlist = Channel.fromPath( crams )
+    .map { it[-1] as String } // get only filename
+    .collectFile( name: "cramlist.txt", newLine: true, sort: true )
+
+    reads = Channel.of( reads ).combine( cramlist )
+
+    STITCH (
+        stitch_input,
+        reads,
+        reference,
+        seed
+    )
```

Comment:
Could you also add a test with the RData synchronisation ?
You could do that by getting the result of the previous test and feed it to the second ?

Reply:
It would also be nice to test without the seed and check for only the presence of the files.

Reply:
Thanks for the comments!

I think better to run it with a different test - first generate the input and then imputing. The current test is a single step imputation. Would you test without the seed only or in addition to the current test?

Reply:
Both imputation in one step and imputation in two would be nice to be tested.
I would put the test without the seed as an addition.